test/unit/int_algolia/scripts/algolia/lib/pageDesignerHelper.test.js

```diff
@@ -0,0 +1,170 @@
+    getRegionDefinitons,
+    isIgnoredAttribute,
+    getRegionContent
+} = require('../../../../../../cartridges/int_algolia/cartridge/scripts/algolia/lib/pageDesignerHelper');
+
+jest.mock('dw/experience/PageMgr', () => {
+    class PageMock {
+        constructor() {
+            this.id = 'mockPage';
+            this.typeID = 'storePage';
+            this.regions = {
+                mockRegion: {
+                    visibleComponents: [{
+                        id: 'mockComponent1',
+                        testAttribute: 'testValue',
+                    },
+                    {
+                        id: 'mockComponent2'
+                    }],
+                },
+            };
+        }
+
+        getRegion(id) {
+            return this.regions['mockRegion'];
+        }
+
+        getAttribute(id) {
+            return this.regions['mockRegion'].visibleComponents[0].testAttribute;
+        }
+
+        getPage(id) {
+            return this;
+        }
+    }
+
+    return {
+        getPage: jest.fn().mockImplementation((id) => {
+            return new PageMock()
+        }),
+    };
+}, {
+    virtual: true
+});
+
+jest.mock('*/cartridge/experience/mockType/storePage.json', () => {
+    const originalModule = jest.requireActual('../../../../../mocks/sfra/experiences/storePage.json');
+    return originalModule;
+}, {
+    virtual: true
+});
+
+jest.mock('*/cartridge/experience/components/storePage.json', () => {
+    const originalModule = jest.requireActual('../../../../../mocks/sfra/experiences/editorialRichText.json');
+    return originalModule;
+}, {
+    virtual: true
+});
+
+describe('isIndexableComponent', () => {
+    test('should return true for searchable components', () => {
+        const component = {
+            searching: {
+                searchable: true
+            }
+        };
+        expect(isIndexableComponent(component)).toBe(true);
+    });
+
+    test('should return false for non-searchable components', () => {
+        const component = {
+            searching: {
+                searchable: false
+            }
+        };
+        expect(isIndexableComponent(component)).toBe(false);
+    });
+
+    test('should return false for components without searching property', () => {
+        const component = {};
+        expect(isIndexableComponent(component)).toBe(false);
+    });
+});
+
+describe('getContainerContent', () => {
+    test('should retrieve content from container correctly', () => {
+        // Set up a mock container and type
+        const container = require('dw/experience/PageMgr').getPage('mockPage');
+
+        const type = 'mockType';
+
+        // Call the function and verify the result
+        const result = getContainerContent(container, type);
+        // Expectations based on the mock data
+
+        // Verify that the correct page was retrieved (due to mock data)
+        expect(result).toBe('testValue testValue testValue testValue testValue testValue');
```

Comment:
Why is there 6 `testValue` extracted here?

Reply:
This function works recursively and We are using `test/mocks/sfra/experiences/storePage.json` and there are 3 regions. Each regions includes 2 components according to PageMock and that is the reason that 6 `testValue`

Reply:
Ah, ok but only the first attribute has the `testValue`, but I see that the mock is always returning this one actually: `this.regions['mockRegion'].visibleComponents[0]`

Maybe the mock can be made more realistic (with a `id`) parameter?

Reply:
Thank you for the hint. I made it a bit more realistic now

Reply:
Unfortunately it's still not realistic. I now understand why we get 6 `testValue`:
- The `page.getRegion()` systematically returns the `mockRegion` region, even though the `storePage` definition doesn't contain any `mockRegion` region
- The `PageMgr.getPage()` systematically returns the same PageMock instance, without taking into account the `container.ID`, so `getContainerContent()` returns the same value for the `richText` component and for the `mockComponent2`

So in the end with indeed have 3 * 2 occurrences, but the mock setup is confusing.

Reply:
Honestly, my purpose was to test if the recursive function was working correctly here. It is not important what values it returns. But sure, I can still spend more time creating a mock page to handle these types of scenarios.